packages/volto/src/components/manage/Blocks/Listing/getAsyncData.js

```diff
@@ -2,12 +2,15 @@ import { getQueryStringResults } from '@plone/volto/actions';
 import { resolveBlockExtensions } from '@plone/volto/helpers';
 
 export default function getListingBlockAsyncData(props) {
-  const { data, path, id, dispatch, blocksConfig, content } = props;
+  const { data, path, location, id, dispatch, blocksConfig, content } = props;
 
   const { resolvedExtensions } = resolveBlockExtensions(data, blocksConfig);
 
   const subrequestID = content?.UID ? `${content?.UID}-${id}` : id;
 
+  // retrieve the current page from the location querystring
+  const currentPage = location?.query?.split(/page=(\d+)/)[1] || 1;
```

Comment:
Maybe use querystring to parse the query? This feels dirty, we already import the qs package.

Reply:
> Maybe use querystring to parse the query? This feels dirty, we already import the qs package.

Fake news, don't review a draft pr or your comments might end up stale as can be seen in this commit 
https://github.com/plone/volto/pull/5689/commits/ad68293b65870772bd38fd008ef01103f197e59f